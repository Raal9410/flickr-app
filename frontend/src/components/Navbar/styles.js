import styled from 'styled-components';

const NavbarContainer = styled.div`
  width: 90vw;
  display: flex;
  flex-direction: row;
  align-items: center;
  align-self: flex-end;

  @media (min-width: 600px) {
    width: 85vw;
  display: flex;
  flex-direction: row;
  align-items: center;
  align-self: flex-end;
  margin-left: 100px;
  }
`;

const HomeIcon = styled.img`
  width: 55px;
  height: auto;

  @media (min-width: 600px) {
    width: 50px;
    height: auto;
  }
`;

const Input = styled.input`
  width: 90vw;
  height: 50px;
  border: 1px solid black;
    border-radius: 5px;

  @media (min-width: 600px) {
    width: 100%;
    height: 40px;
    border: 1px solid black;
    border-radius: 5px;
  }
`;

const Button = styled.button`
  width: 50px;
  height: 50px;
  margin: 5px;
  padding: 0.25em 1em;
  border: 2px solid black;
  border-radius: 3px;
  background: white;
  font-size: 0;
  object-fit: contain;
  @media (min-width: 600px) {
    width: 100px;
    height: 40px;
    color: gray;
    margin: 1em;
    font-size: 15px;
    padding: 0.25em 1em;
    border: 2px solid black;
    border-radius: 3px;
    background: white;
  }
`;

const Lens = styled.img `
width: 25px;
height: 25px;
    @media(min-width: 600px){
        display: none;
    }
`

export { NavbarContainer, HomeIcon, Input, Button, Lens };
