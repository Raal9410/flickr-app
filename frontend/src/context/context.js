import { useState, createContext } from 'react';
import API from '../config/API';

const useFlickrApi = () => {
  const [state, setState] = useState({
    loading: false,
    error: null,
    images: [],
  });

  const fetchPublicFeed = async () => {
    setState(prevState => ({
      ...prevState,
      loading: true,
    }));

    const {
      data: { results },
    } = await API.get('/public');

    setState(prevState => ({
      ...prevState,
      loading: false,
      images: results,
    }));
  };

  const fetchTagFeed = async tag => {
    setState(prevState => ({
      ...prevState,
      loading: true,
    }));

    const {
      data: { results },
    } = await API.get(`/filtered/${tag}`);

    setState(prevState => ({
      ...prevState,
      loading: false,
      images: results,
    }));
  };

  return { ...state, fetchPublicFeed, fetchTagFeed };
};

const Context = createContext(null);

export { useFlickrApi, Context };
