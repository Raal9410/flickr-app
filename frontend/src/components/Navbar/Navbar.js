import React, { useState, useContext } from 'react';
import { Context } from '../../context/context';

import home from './assets/home.png';
import lens from './assets/lens.png';

import { NavbarContainer, HomeIcon, Input, Button, Lens } from './styles';

const Navbar = () => {
  const [tag, setTag] = useState('');
  const { fetchTagFeed, fetchPublicFeed } = useContext(Context);

  const fetchImages = e => {
    e.preventDefault();
    fetchTagFeed(tag);
  };

  const changeTag = e => setTag(e.target.value);
  return (
    <NavbarContainer>
      <HomeIcon src={home} onClick={fetchPublicFeed} />
      <Input type="text" onChange={changeTag} placeholder="What are you looking for?" />
      <Button onClick={fetchImages}>
        Search
        <Lens src={lens} />
      </Button>
    </NavbarContainer>
  );
};

export default Navbar;
