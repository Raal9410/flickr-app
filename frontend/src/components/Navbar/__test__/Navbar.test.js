import React from 'react';
import { render } from 'enzyme';

import Navbar from '../Navbar';
import { Context } from '../../../context/context';

describe('<Navbar />', () => {
  describe('Rendering', () => {
    it('matches the snapshot', () => {
      const data = {};
      const component = render(
        <Context.Provider value={data}>
          <Navbar />
        </Context.Provider>,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
