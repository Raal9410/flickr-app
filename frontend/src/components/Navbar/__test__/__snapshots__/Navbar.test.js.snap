// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> Rendering matches the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-AxjAm dAzeQN",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-AxirZ bnnxjp",
          "src": "home.png",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-AxiKw hgimLr",
            "placeholder": "What are you looking for?",
            "type": "text",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "sc-AxhCb bJlebg",
            },
            "children": Array [
              Object {
                "data": "Search",
                "next": Object {
                  "attribs": Object {
                    "class": "sc-AxhUy dhHEeR",
                    "src": "lens.png",
                  },
                  "children": Array [],
                  "name": "img",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "src": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "src": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              Object {
                "attribs": Object {
                  "class": "sc-AxhUy dhHEeR",
                  "src": "lens.png",
                },
                "children": Array [],
                "name": "img",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "data": "Search",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "src": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "src": undefined,
                },
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "placeholder": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "placeholder": undefined,
            "type": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "src": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "src": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-AxiKw hgimLr",
          "placeholder": "What are you looking for?",
          "type": "text",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-AxhCb bJlebg",
          },
          "children": Array [
            Object {
              "data": "Search",
              "next": Object {
                "attribs": Object {
                  "class": "sc-AxhUy dhHEeR",
                  "src": "lens.png",
                },
                "children": Array [],
                "name": "img",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "src": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "src": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
            Object {
              "attribs": Object {
                "class": "sc-AxhUy dhHEeR",
                "src": "lens.png",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "data": "Search",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "src": undefined,
              },
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-AxirZ bnnxjp",
            "src": "home.png",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "src": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "placeholder": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "placeholder": undefined,
          "type": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-AxhCb bJlebg",
        },
        "children": Array [
          Object {
            "data": "Search",
            "next": Object {
              "attribs": Object {
                "class": "sc-AxhUy dhHEeR",
                "src": "lens.png",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "src": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
          Object {
            "attribs": Object {
              "class": "sc-AxhUy dhHEeR",
              "src": "lens.png",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "data": "Search",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "src": undefined,
            },
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-AxiKw hgimLr",
            "placeholder": "What are you looking for?",
            "type": "text",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "sc-AxirZ bnnxjp",
              "src": "home.png",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "src": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "placeholder": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "placeholder": undefined,
            "type": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
